src/main/java/bio/terra/app/configuration/CacheConfiguration.java

```diff
@@ -0,0 +1,83 @@
+package bio.terra.app.configuration;
+
+import com.google.common.cache.CacheBuilder;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CachingConfigurer;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.cache.concurrent.ConcurrentMapCache;
+import org.springframework.cache.concurrent.ConcurrentMapCacheManager;
+import org.springframework.cache.interceptor.CacheErrorHandler;
+import org.springframework.cache.interceptor.CacheResolver;
+import org.springframework.cache.interceptor.KeyGenerator;
+import org.springframework.cache.interceptor.SimpleCacheErrorHandler;
+import org.springframework.cache.interceptor.SimpleCacheResolver;
+import org.springframework.cache.interceptor.SimpleKeyGenerator;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.lang.NonNull;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Configures a cache to be used by Spring.  It likely makes sense to use a cache per endpoint.
+ */
+@EnableCaching
+@Configuration
+public class CacheConfiguration implements CachingConfigurer {
+
+    public static final String SPRING_CACHE_MANAGER = "springCacheManager";
+
+    // Use these strings to reference CacheProperties
+    public static final String STATUS_PROP = "STATUS";
+
+    public enum CacheProperties {
+        STATUS(30L, TimeUnit.SECONDS, null);
```

Comment:
Is 30s too long to wait to get fresh status? The thing to consider here is, if jade is not working, and that problem would be resolved by restarting, would it be OK to wait 30s to find out?

Another side effect is that our "downtime" stats will reflect the cached value, so if jade is unavailable for 20s at a time we might never know.

One last thought, if the result is "failed", maybe we don't want to cache that value and always fetch a fresh value, so we can report the system being back up as soon as it occurs.

Reply:
I'm not sure about not caching if it's failed but fair point about the time being too long.  I'll reduce to 5 seconds